core/src/main/java/org/bptlab/cepta/patterns/StaysInStationPattern.java

```diff
@@ -0,0 +1,67 @@
+package org.bptlab.cepta.patterns;
+
+import org.apache.flink.streaming.api.windowing.time.Time;
+import org.apache.flink.streaming.api.datastream.DataStreamUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.cep.PatternStream;
+import org.apache.flink.cep.CEP;
+import org.apache.flink.cep.pattern.*;
+import org.apache.flink.cep.pattern.conditions.*;
+import org.bptlab.cepta.models.events.train.LiveTrainDataProtos.LiveTrainData;
+import org.bptlab.cepta.models.events.correlatedEvents.StaysInStationEventProtos.StaysInStationEvent;
+
+import java.util.*;
+
+
+public class StaysInStationPattern {   
+    public static final Pattern<LiveTrainData, ?> staysInStationPattern = 
+      Pattern.<LiveTrainData>begin("arrivesInStation")
+      .where(new SimpleCondition<LiveTrainData>(){
+        @Override
+        public boolean filter(LiveTrainData event) {
+          return event.getStatus() == 3;
+        }
+      })
+      .next("departuresFromStation")
+      .where(new IterativeCondition<LiveTrainData>(){
+        @Override
+        public boolean filter (LiveTrainData incoming, Context<LiveTrainData> context){
+        if (incoming.getStatus() != 4) {
+          return false;
+        }
+        
+        LiveTrainData first = null;
+        try {
+          for (LiveTrainData previous : context.getEventsForPattern("arrivesInStation")){
+            first = previous;
+            break;
+          }
+        } catch (Exception e) {
+          //TODO: handle exception
+        }
+       
+        if (first.getStationId() == incoming.getStationId()) {
+          return true;
+        }
```

Comment:
Why not put the status and id conditions together in the pattern? Use an IterativeCondition and access the previous elements via the context (see https://ci.apache.org/projects/flink/flink-docs-stable/dev/libs/cep.html#individual-patterns for reference).